### PR TITLE
feat: add myList page #1904

### PR DESCRIPTION
**Pull request by lenasokolova**

### CODE REVIEW

File: pages/mylist.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    return {
+      props: {
+        moviesData: myList ? myList.results : [],
+      },
+    };
+  } catch (e) {
+    console.error(e);
+    return { props: {} };
+  }
+}
```
Comment (rast-mk1) [inline pages/mylist.js:21]: Try with these changes:
```suggestion
export async function getServerSideProps({ req }) {
  try {
    const myList = await getMyList(req);

    return {
      props: {
        moviesData: myList ? myList : [],
      },
    };
  } catch (e) {
    console.error(e);
    return { props: {} };
  }
}
```

**Review by rast-mk1 — APPROVED** [review]

File: app/components/Pages/MyListPage/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  SearchStyle,
+  ErrorWrapper,
+  NoMovieError,
+  TitleWrapper,
+  MyListTitle,
+} from './MyListPage.styles';
+
+export default function MyListPage({ moviesData = [] }) {
+  const [searchQ, setSearchQ] = useState('');
+  console.log(moviesData);
```
Comment (weronika299) [inline app/components/Pages/MyListPage/index.js:17]: Could you remove this console.log?

File: app/components/Pages/MyListPage/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        </NoMovieError>
+      </ErrorWrapper>
+    );
+  }
+  const filteredList = searchQ
+    ? moviesData.filter((movie) => movie.title?.toLowerCase().includes(searchQ.toLowerCase()))
+    : moviesData;
+
+  return (
+    <>
```
Comment (weronika299) [inline app/components/Pages/MyListPage/index.js:33]: Is this empty tag necessary?

Comment (lenasokolova) [reply app/components/Pages/MyListPage/index.js:33]: you're right. Here's only one parent so JSX tag is redundant here :)

File: app/components/Pages/MyListPage/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            <Input
+              id="search"
+              placeholder="Search the movies in this category..."
+              onChange={(event) => setSearchQ(event.target.value)}
+              value={searchQ}
+            />
+          </SearchStyle>
+        </SearchDiv>
+        <TitleWrapper>
+          <MyListTitle></MyListTitle>
```
[the remark this answers is not in the document]
Comment (lenasokolova) [reply app/components/Pages/MyListPage/index.js:46]: oops I'll fix that

File: app/components/Pages/MyListPage/index.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  SearchDiv,
+  SearchStyle,
+  ErrorWrapper,
+  NoMovieError,
+  TitleWrapper,
+  MyListTitle,
+} from './MyListPage.styles';
+
+export default function MyListPage({ moviesData = [] }) {
+  const [searchQ, setSearchQ] = useState('');
```
Comment (marcelina-hasiak) [inline app/components/Pages/MyListPage/index.js:16]: const [query, setQuery] = useState('') <- I would rename these two

File: pages/mylist.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  return <MyListPage moviesData={moviesData} />;
+}
+
+export async function getServerSideProps({ req }) {
+  try {
+    const myList = await getMyList(req);
+
+    return {
+      props: {
+        moviesData: myList ? myList : [],
```
Comment (marcelina-hasiak) [inline pages/mylist.js:14]: Our handler already returns empty array if there's no list. Check api/myList.js file

**Review by marcelina-hasiak — CHANGES_REQUESTED** [review]

File: pages/mylist.js (hunk 6 above)
Comment (lenasokolova) [reply pages/mylist.js]: yeah, you're right! :)

**Pull request merged**